### PR TITLE
Fb auth issues

### PR DESCRIPTION
**Pull request by emilyb7**
better error handling for facebook api!

\+ merged in from #240 : use library to generate unique ids for new users

### CODE REVIEW

**Comment by emilyb7** [comment]
@shiryz if you're happy, we could close #240 and just do this one, as they work well together...

File: server/database/get-user.js [hunk 1]
```diff
@@ -1,12 +1,15 @@
 const dbClient = require('./db_connection.js');
+const uuidV1 = require('uuid/v1');
 
 const findUser = `
 SELECT * from users
 WHERE username = $1;`;
 
+const newId = uuidV1();
```
Comment (des-des) [inline server/database/get-user.js]: Urm, what is this?

Each time you start the server you create this only once. Does it represent the server (what is going on)?

File: server/database/get-user.js [hunk 2]
```diff
@@ -20,7 +23,7 @@ module.exports = (facebookId, callback) => {
       return callback(null, result.rows[0].user_id);
     }
 
-    dbClient.query(createUser, [ facebookId, ], (insertErr, newUser) => {
+    dbClient.query(createUser, [ newId, facebookId, ], (insertErr, newUser) => {
```
Comment (des-des) [inline server/database/get-user.js]: What is `newId`? Why is it called `newId`? You are not using it look up users, only create them.. Not really sure what is going on

**Comment by des-des** [comment]
@shiryz @emilyb7 We talked about this, fixes on the way I expect

File: server/database/db_build.sql [hunk 3]
```diff
@@ -18,8 +17,8 @@ CREATE TABLE goals (
 
 CREATE TABLE ratings (
   rating_id SERIAL PRIMARY KEY NOT NULL,
-  user_id INTEGER,
-  goal_id INTEGER,
+  user_id TEXT,
```
Comment (des-des) [inline server/database/db_build.sql]: Not going to block this as not in this pull, but these should be foreign keys

**Pull request merged**